### PR TITLE
Theme attempt 3 using awdark & awlight

### PR DESCRIPTION
**Pull request by windymilla**
Three themes: default, dark and light.

Default is now platform default (aqua/vista/default), not always the theme named "default".

Line numbers widget, text widget, and readonly text in checker dialogs all try to use a sensible color for the theme - they are not themed widgets, so it has to be done manually. Text widgets are either the default, or black-on-white or white-on-black. Line number widget copies text and background colors from the themes colors for a button.

### CODE REVIEW

**Comment by tangledhelix** [comment]
I also see the "invisible cursor" issue in Dark theme on Ubuntu.

**Comment by srjfoo** [comment]
> At least on Mac, when using Dark theme the cursor is not visible. I can move around and it seems the cursor is working as expected - I just can't see it.
> 
> (edited to add - to be clear, I did restart GG to make sure the theme was fully loaded.)

I think that's probably a function of your system theme. On the Dark theme, I could see the cursor just fine, but if I switched to the light theme, I couldn't, other than a telltale blink at the left side of a selection if I selected some text. If I switched my system to the light theme, that flipped: I could see the cursor on the light theme, but not the dark.

Default behaves very weirdly. If the last theme I had set was dark, the main text window stays dark (assuming light system theme); if the last theme I had set was light, it shows light.

**Comment by srjfoo** [comment]
Further comments (with a `master` instance up for visual comparison).

Screenshot shows system theme (Dark), main GG window, search-results window, and settings pane (Default).

image

The previous setting for the Theme was `Light`, and this is what I get switching to `Default`: The outer framework switches to the default (i.e. what I see when I bring up an instance of `master`). 

If I now switch to `Dark` in `Settings`, and then back to `Default`, the display retains the dark-background text windows. (I hope that does a good job of describing what I'm seeing.)

**Comment by srjfoo** [comment]
@windymilla, what do you see on Windows. We already know that themes behave differently on Mac and Windows.

So far, I haven't seen anything in the three theming PRs that is even as good as the default I get in `master`, but that doesn't give Windows users the ability to set a dark theme (Since Windows apparently gives one a light theme even if one has the system set to a dark theme).

Of the three PRs so far, the two I'm fondest of have been `solar` and `superhero` (particularly the latter) in the first one, but that PR doesn't give us any kind of default option, so is not ideal because of that.

**Comment by srjfoo** [comment]
I forgot the bit about restarting GG (even though I read that bit (🙄). Thanks for the reminder @windymilla. Yes, if I restart GG, after switching from the light theme to `Default`, it goes back to what I expect that to look like. (I've gotten out of the habit of needing to restart programs for something to take effect, and it no longer feels intuitive!

**Comment by windymilla** [comment]
@srjfoo - ready for a proper look now, thanks.

**Comment by windymilla** [comment]
> At least on Mac, when using Dark theme the cursor is not visible. I can move around and it seems the cursor is working as expected - I just can't see it.

Thanks - fixed in latest commits, I think.

**Review by srjfoo — APPROVED** [review]

File: src/guiguts/maintext.py [hunk 1]
```diff
@@ -78,11 +79,15 @@ def __init__(
     ) -> None:
         self.textwidget = text_widget
         self.font = tk_font.nametofont(self.textwidget.cget("font"))
-        self.offset = 5
+        self.offset = 10
         # Allow for 5 digit line numbers
```
[the remark this answers is not in the document]
Comment (windymilla) [reply src/guiguts/maintext.py:83]: No, it's the offset from the edge of the line numbers widget (roughly the gap between the line numbers and the main text). There used to be a border around the line numbers widget, but when I changed the background color it left a thin border of the wrong color all the way round. When I removed the border (called highlightthickness later on in the file) the numbers were too close to the text, so I upped the offset.

**Pull request merged**